### PR TITLE
[UNR-3526] Allow specifying port from `open ip:port`

### PR DESCRIPTION
**Pull request by BenNaccarato**
#### Description
Add ReceptionistPort getters and setters, and include setting the port in SetupFromURL

#### Release note
Adds support to specify a `receptionistPort` in the `URL` when using client travel.

#### Tests
No tests, but I did the equivalent in Stavka, and it worked

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h [hunk 1]
```diff
@@ -270,5 +273,7 @@ class FReceptionistConfig : public FConnectionConfig
 		}
 	}
 
+	void SetReceptionistPort(const uint16 port) { ReceptionistPort = port; }
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h:276]:
```suggestion
	void SetReceptionistPort(const uint16 Port) { ReceptionistPort = Port; }
```

**Comment by mironec** [comment]
Updated this so it passes tests, and modified them slightly so they test this new functionality as well.

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialConnectionManagerTest.cpp [hunk 2]
```diff
@@ -11,26 +11,26 @@ class FTemporaryCommandLine
...
-    {
-    	if (OldCommandLine.IsEmpty())
-    	{
-    		OldCommandLine = FCommandLine::GetOriginal();
-    		FCommandLine::Set(*NewCommandLine);
-    		bDidSetCommandLine = true;
-    	}
-    }
+	{
+		if (OldCommandLine.IsEmpty())
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialConnectionManagerTest.cpp:15]: Nice fix on the spaces but this still seems off in the GH UI.

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialConnectionManagerTest.cpp [hunk 3]
```diff
@@ -118,7 +118,7 @@ CONNECTIONMANAGER_TEST(SetupFromURL_Receptionist_Localhost)
 {
 	// GIVEN
 	FTemporaryCommandLine TemporaryCommandLine("");
-	const FURL URL(nullptr, TEXT("127.0.0.1"), TRAVEL_Absolute);
+	const FURL URL(nullptr, TEXT("127.0.0.1:777"), TRAVEL_Absolute);
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialConnectionManagerTest.cpp:121]:
```suggestion
	const FURL URL(nullptr, TEXT("127.0.0.1:7777"), TRAVEL_Absolute);
```
Nit 7777 is the actual runtime port

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialConnectionManagerTest.cpp:121]: Ah I see you're using a different one and then testing the default. Potentially ignore this then.

**Pull request merged**